Move sequential partition utils to fx/passes/utils

cccclai opened the pull request:
Summary: `find_sequential_partitions` is fairly generic. Let's move it to `fx/utils` so it can be used by non quantization related work

Test Plan: CI

Differential Revision: D48664627

jerryzh168 commented on torch/ao/quantization/quantizer/x86_inductor_quantizer.py:
```diff
@@ -31,6 +30,7 @@
     QuantizationConfig,
 )
 from torch.fx import Node
+from torch.fx.passes.sequential_partition_utils import find_sequential_partitions
```
should this just be moved to source_matcher_utils?

jerryzh168 commented on torch/fx/passes/utils/source_matcher_utils.py:
```diff
@@ -141,3 +145,100 @@ def check_subgraphs_connected(subgraph1: SourcePartition, subgraph2: SourceParti
+    for candidate in fusion_candidates:
+        if _partitions_sequential(candidate):  # type: ignore[arg-type]
+            fused_partitions.append(candidate)
+    return fused_partitions
```
please fix the lint

jerryzh168 reviewed (changes requested):
please fix lint

cccclai committed:
Move sequential partition utils to fx/passes/utils (#108605)

Summary:

`find_sequential_partitions` is fairly generic. Let's move it to `fx/utils` so it can be used by non quantization related work

Test Plan: CI

Differential Revision: D48664627

jerryzh168 commented on torch/fx/passes/utils/__init__.py:
```diff
@@ -1 +1,9 @@
+
+
+
+__all__ = [
```
should this be added to the `__all__` list in source_matcher_utils as well?
